src/atoms/CustomHtml/CustomHtml.js

```diff
@@ -1,34 +1,99 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import ReactDOM from 'react-dom';
 import whitelist from '../../utils/whitelist.json';
 
 class CustomHtml extends React.Component {
   constructor(props) {
     super(props);
-
+    this.myRef = React.createRef();
     this.state = {
-      html: ''
+      html: '',
+      loaded: false
     };
   }
 
   componentDidMount() {
+    const { html, safeScripts } = this.parseHtml({
+      ...this.props.nodeData.attrs,
+      whitelist: whitelist.approved
+    });
+
+    // eslint-disable-next-line
+    const node = ReactDOM.findDOMNode(this.myRef.current);
+    if (!this.state.loaded) {
+      safeScripts.forEach((script) => {
+        this.loadScript(script.src, node);
+      });
+    }
+
     this.setState({
-      html: cleanHtml(this.props.nodeData.attrs)
+      html,
+      loaded: true
     });
   }
 
+  loadScript(src, node) {
+    let tag = document.createElement('script');
+
+    tag.type = 'text/javascript';
+    tag.async = false; // Load in order
+    tag.src = src;
+    node.appendChild(tag);
+    return tag;
+  }
+
+  /**
+   * Removes all scripts with an external source from the html and returns the whitelisted ones as an array of Nodes
+   * @param {html: string, fallback_url: string, whitelist: Array}
+   * @returns {html: string, safeScripts: Node[]}
+   */
+  parseHtml({ html, fallback_url, whitelist }) {
+    const whitelistRegex = new RegExp(whitelist.join('|'));
+    const element = this.htmlStringToElement(html);
+    let safeHtml, scripts, safeScripts, hasIframe;
+
+    if (!element) return { html: '', safeScripts: [] };
+
+    scripts = Array.from(element.querySelectorAll('script[src]'));
```

Comment:
What happens if there is a script tag without a src attribute?

Reply:
Roughly the same thing that was happening before. If there is a script without a source the script is placed in an iframe and the source is set to the fallback_url if it is in the whitelist, or without a src if it isn't whitelisted.